### ISSUE TITLE
roundtrip for proleptic_gregorian calendar wrong for reference dates before Julian/Gregorian transition

### ISSUE THREAD

**Issue by jswhit** [Unidata/cftime#189]
```python
import cftime as cf
from cftime import DatetimeProlepticGregorian as datetime
dt1 = datetime(2020, 4, 24, 16, 15, 10)
print(dt1)
units='days since -4713-01-01 12:00'
cal='proleptic_gregorian'
print(cf.num2date(cf.date2num(dt1, units, cal), units, cal))

2020-04-24 16:15:10
2020-03-19 16:15:10
```

For this example, it's off by 38 days.

Correct result is obtained if reference date year changed to 1583.

**Comment by jswhit** [comment]
Actually, it works for reference date years back to -100.  Suspect it has something to do with the calculation of leap years for negative years.  

It's a bug in _is_leap - fix is included in PR #188

**Comment by jswhit** [comment]
closed by PR #188
